### DISCUSSION TITLE
Is LdapConnection threadsafe?

### DISCUSSION THREAD

**Discussion by FinHorsley** [dsbenghe/Novell.Directory.Ldap.NETStandard#237] [General]
Apologies if this has been answered before, I've had a look at the discussions but couldn't see a definitive answer for this aside from https://github.com/dsbenghe/Novell.Directory.Ldap.NETStandard/discussions/235. Is the `LdapConnection` threadsafe?

Related to this, can a single `LdapConnection` be used for multiple `Connect` operations, assuming the same ipAddress and port are used for each `Connect` call on a re-used ldapConnection? I'd assume we wouldn't want to connect as part of the pool, as this would open a long-lived connection to the ldapServer which may end/error (etc) if the connection isn't used by the pool for a period of time.

For example, our application can be configured to talk to n number of different ldapConnections. Currently, we create a new `ldapConnection` for each ldap request, specifying the ipaddress/port as part of the `Connect` call. If we were to update our code to pool the connections, would we need to create a new ldapConnection for each ipaddress/port combination in the pool to prevent issues?

from documentation
> An LdapConnection object is not connected on construction and can only be connected to one server at one port.

**Comment by dsbenghe** [comment]
A LdapConnectin should not be accessed from multiple threads - this is business as usual. Most of the "client' libraries connecting to a server are working like this.

This lib comes with a feature of multiplexing and reusing one underlaying ldap tcp/ip connection by multiple consumers via LdapConnection Clone method. This functionality has existed since the original Novell implementation but I haven't used it and I can not say if it works and/or it is stable and it is not covered by existing tests. 

Give it a go and report how  it goes. Maybe, when I have time, I will add tests for it.
